### PR TITLE
docs: add defineConfig helper to match web-infra-dev/rspack#2267

### PR DESCRIPTION
**Pull request by so1ve**

### CODE REVIEW

File: docs/en/config/index.mdx [hunk 1]
```diff
@@ -18,17 +18,16 @@ module.exports = {
 
 ### Type Checking
 
-`rspack.config.js` is a JavaScript file, you can use JSDoc to enable the IDE's Intellisense and TypeScript type checking.
+`rspack.config.js` is a JavaScript file, you can use the `defineConfig` helper to enable the IDE's Intellisense and TypeScript type checking.
 
 ```js title="rspack.config.js"
-// @ts-check
```
Comment (h-a-n-a) [inline docs/en/config/index.mdx:24]: We want to keep our example as neat as possible, you may start a new alternative down below with `defineConfig`.

File: docs/en/config/index.mdx [hunk 2]
```diff
@@ -18,17 +18,16 @@ module.exports = {
...
+import { defineConfig } from '@rspack/cli';
```
Comment (h-a-n-a) [inline docs/en/config/index.mdx:24]: Please stick to `require` semantic as extension `js` requires additional setup and `mjs` is currently not supported. Be sure to include `// @ts-check` if type checking is necessary.

File: docs/en/config/index.mdx (hunk 1 above)
Comment (so1ve) [reply docs/en/config/index.mdx:24]: Oops, sorry. I will revert this later

File: docs/zh/config/index.mdx [hunk 3]
```diff
@@ -18,17 +18,16 @@ module.exports = {
 
 ### 配置类型检查
 
-`rspack.config.js` 是一个 JavaScript 文件，你可以通过 JSDoc 来启用 IDE 的智能补全和 TypeScript 类型检查。
+`rspack.config.js` 是一个 JavaScript 文件，你可以通过 `defineConfig` 工具函数来启用 IDE 的智能补全和 TypeScript 类型检查。
 
 ```js title="rspack.config.js"
-// @ts-check
```
Comment (h-a-n-a) [inline docs/zh/config/index.mdx:24]: You may need to change this too.

File: docs/zh/config/index.mdx [hunk 4]
```diff
@@ -29,7 +29,21 @@ const config = {
     main: './src/index.js',
   },
 };
+module.exports = config;
+```
+
+你也可以使用 `defineConfig` 工具函数：
```
Comment (h-a-n-a) [inline docs/zh/config/index.mdx:35]: Would you add some explanations what this helper is doing? Some users may not familiar with this pattern.

**Comment by so1ve** [comment]
@h-a-n-a Can you review again? Thanks!

File: docs/zh/config/index.mdx (hunk 4 above)
Comment (so1ve) [reply docs/zh/config/index.mdx]: Added

**Review by h-a-n-a — APPROVED** [review]

**Comment by so1ve** [comment]
Sorry about so many mistakes :-( I'll do better next time

**Pull request merged**